src/pybind/mgr/dashboard/frontend/src/app/ceph/pool/pool-form/pool-form.component.ts

```diff
@@ -411,20 +411,24 @@ export class PoolFormComponent implements OnInit {
     }
   }
 
-  private replicatedPgCalc(pgs): number {
+  private replicatedPgCalc(pgs: number): number {
     const sizeControl = this.form.get('size');
     const size = sizeControl.value;
     if (sizeControl.valid && size > 0) {
       return pgs / size;
     }
+
+    return undefined;
```

Comment:
This is due to `noImplicitReturns`.
I'm OK with disabling it, if requested.

But I think the code is actually more readable and you will never forget to return something or miss that its not returning when reviewing.

Also, I prefer the current code, it might have more lines but its more readable.

Reply:
We should vote for it during the next stand up :)
I'm for disabling it, as how a method/function works should be known.

Reply:
> But I think the code is actually more readable and you will never forget to return something or miss that its not returning when reviewing.

FWIW, I too am in favor of better readability. Keep in mind that we should write code that others could quickly get familiar with in order to lower the barrier for new contributors (and for reviewers, FWIW).

I suggest to keep the check for implicit returns - this also improves the readability and avoids ambiguity.

Reply:
I tend to be more strict as well. I like the `noImplicitReturns` rule and I think (and hope) it will eventually lead to code that looks like it was written in a strictly typed language. That is, when declared to return a number, will return a number. Taking the example, it'd look like this:

```typescript
private replicatedPgCalc(pgs: number): number {
  const sizeControl = this.form.get('size');
  const size = sizeControl.value;
  if (sizeControl.valid && size > 0) {
    return pgs / size;
  }
  return 0; // or: -1   <---
}
```

As alternative, the method/function can throw an exception which can be caught by the caller.

This looks way more natural to me than writing `return undefined`, which by the way, feels unnatural to me for two reasons:

1. The method declared a number to be returned but returned nothing
2. `undefined` is the return value for attributes of objects which aren't there or method/function arguments that haven't been provided. I'd prefer to use `null` if necessary, but would even try to avoid that by either returning a variable of the declared type or raising an exception.

Either way, **the way to improve in that regard is to keep `noImplicitReturns`** and eventually write code that doesn't feel so unnatural. But as a first step I wouldn't require the code to be changed this way in this PR, as this is the current behavior the code and the PR is really big enough, even without changing its behavior.

Reply:
Any changes can me made in a follow up PR to get it in quickly.